### DISCUSSION TITLE
How to find out if cursor is over an element

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2492] [Q&A] [answered]
I am interested in finding out when the cursor moves over an element in iModel.  This is the same event which causes the Viewer to highlight the element and display its tooltip.

Is there an event I can subscribe to (or another way to find out) if cursor is over an element.  I want to show certain info and possibly trigger certain user actions (with a popup toolbar).  I suppose I can trigger my actions from getTooltip() but I believe it would be a better design if I can separately subscribe to such an event (if it exists) and handle it independent of tooltip.

**Comment by pmconne** [comment, marked as answer]
[Viewport.onFlashedIdChanged](https://www.itwinjs.org/reference/imodeljs-frontend/views/viewport/#onflashedidchanged).
Note: the flashed Id can change frequently, as the user moves the mouse cursor around. You will probably want to impose a delay before you start assuming the user is interested in interacting with whatever happens to currently be flashed.

**Reply by ovaisa** [reply]
Thank you Paul, this is exactly what I was looking for, works out perfectly.  Thanks for the note on frequent changing, I was able to preempt the issues this would have caused.
